src/app/Http/Controllers/ApiController.php

```diff
@@ -126,12 +129,79 @@ public function updateAvailability(Request $request, Location $location) {
     public function armorvax(Request $request) {
         $data = $request->all();
         $data_string = json_encode($data, JSON_PRETTY_PRINT);
-        $path = sprintf('armorvax/%s/u%d_%s.json', config('app.env'), $request->user()->id, date('Y-m-d_his'));
+        $data_collection = collect(json_decode($data_string));
 
-        // Write the full JSON data to a file on S3
-        \Storage::disk('s3')->write($path, $data_string);
+        /**
+         * First things first: delete all the existing availabilities
+         * associated with the ArmorVax sites.
+         */
+        Location::locationsWithSource('ArmorVax')->each(
+        function ($location, $location_key) {
+            $location->availabilities->each(function ($availability, $key) {
+                # Delete the availabilities associated with each of the
+                # ArmorVax sites already in the database.
+                $availability->delete();
+            });
+        });
 
-        // echo back the same json that was posted
+        /**
+         * For each of the entries in the posted JSON array ...
+         */
+        $data_collection->each(
+            function ($location, $location_key) {
+                /**
+                 * ... For each of the locations in each of the entries ...
+                 */
+                collect($location->AvailibilitiesByLocation)->each(
+                    function ($availability, $availability_key) {
+                        $name = $availability->LocationName;
+                        $address = Address::standardize($availability->LocationAddress->AddressLine1);
+                        $address2 = Address::standardize($availability->LocationAddress->AddressLine2);
+                        $city = $availability->LocationAddress->City;
+                        $state = $availability->LocationAddress->State;
+                        $zip = $availability->LocationAddress->Zip;
+                        $county = $availability->LocationAddress->County;
+                        $booking_link = "https://app.armorvax.com";
+                        $source = LocationSource::firstOrCreate(['name' => 'ArmorVax'])->id;
```

Comment:
It should be safe to use `LocationSource::where('name','ArmorVax')->id` because we have that included in our seeder. But it may be worth sticking with what you have here just in case it were to get delete somehow.

Reply:
Better safe than sorry!